File: app/src/main/java/br/com/devmarques/picpaytestes/Adapter/AdapterTransacoes.java
package br.com.devmarques.picpaytestes.Adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import br.com.devmarques.picpaytestes.Dados.Cartao;
import br.com.devmarques.picpaytestes.Dados.Transacoes;
import br.com.devmarques.turismo.picpaytestes.R;
import de.hdodenhof.circleimageview.CircleImageView;

/**
 * Created by Roger on 12/11/2017.
 */

public class AdapterTransacoes extends RecyclerView.Adapter<AdapterTransacoes.ViewHolder> {

    ArrayList<Transacoes> transacoesList = new ArrayList<Transacoes>();
    Context context;
    Transacoes transacoes;
    LinearLayout aprovadoOUnao;

    public AdapterTransacoes(ArrayList<Transacoes> transacoesList, Context context) {
        this.transacoesList = transacoesList;
        this.context = context;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.rowtransacoes, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(AdapterTransacoes.ViewHolder holder, int position) {

        holder.nome.setText("Transferido para: "+transacoesList.get(position).getNome()+".");
        holder.transacaoNum.setText("Transação Nº: "+transacoesList.get(position).getTransacaoNum()+".");
        holder.valor.setText("Valor da Transferência R$:"+transacoesList.get(position).getValor()+".");
        holder.aprovado.setText("Aprovado: "+transacoesList.get(position).getAprovada()+".");
        holder.cartaousadoEbandeira.setText("Cartão usado final: "+transacoesList.get(position).getCardNameandNumber()+".");

        retornacor(transacoesList.get(position).getStatus());


        Picasso.with(context)
                .load(transacoesList.get(position).getFotoperfil())
                .resize(300, 300)
                .centerCrop()
                .into(holder.circleImageView);

    }

    @Override
    public int getItemCount() {
        return transacoesList.size();
    }

    public void retornacor(String value){
        if (value.equals("false")){
            aprovadoOUnao.setBackgroundResource(R.drawable.radiusv);
        }
    }


    public class ViewHolder extends RecyclerView.ViewHolder {

         CircleImageView circleImageView;

         TextView transacaoNum, nome, valor,aprovado, cartaousadoEbandeira;

        public ViewHolder(View itemView) {
            super(itemView);
            aprovadoOUnao = itemView.findViewById(R.id.statustransacao);
            circleImageView = itemView.findViewById(R.id.fotouserTransacao);
            transacaoNum = itemView.findViewById(R.id.numeroTransacao);
            nome = itemView.findViewById(R.id.nomedaPessoaTransferida);
            valor = itemView.findViewById(R.id.valordaTransferencia);
            aprovado = itemView.findViewById(R.id.aprovado);
            cartaousadoEbandeira = itemView.findViewById(R.id.cartaousadotransferencia);



        }
    }
}

File: app/src/main/java/projetosestudos/felipe/com/trabalhe_conosco_mobile_dev/adapter/AdapterTransacoes.kt
package projetosestudos.felipe.com.trabalhe_conosco_mobile_dev.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.bumptech.glide.Glide
import de.hdodenhof.circleimageview.CircleImageView
import kotlinx.android.synthetic.main.adapter_transacoes.view.*
import projetosestudos.felipe.com.trabalhe_conosco_mobile_dev.R
import projetosestudos.felipe.com.trabalhe_conosco_mobile_dev.model.AllTransactions

class AdapterTransacoes(context: Context, listaTransactions: ArrayList<AllTransactions>) : RecyclerView.Adapter<AdapterTransacoes.MyViewHolder>() {

    private var mContext: Context? = null
    private var mListTransactions: ArrayList<AllTransactions>? = null

    init {
        mContext = context
        mListTransactions = listaTransactions
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.adapter_transacoes, parent, false))
    }

    override fun getItemCount(): Int {
        return mListTransactions!!.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val transaction = mListTransactions!![position]

        Glide.with(mContext).load(transaction.image).into(holder.mImageUser)
        holder.mTextTransaction!!.text = "Você pagou ${transaction.username}"
        holder.mTextValue!!.text = "R$${transaction.timeTransaction}"
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var mImageUser: CircleImageView? = null
        var mTextTransaction: TextView? = null
        var mTextValue: TextView? = null

        init {
            mImageUser = itemView.profile_image
            mTextTransaction = itemView.textTransaction
            mTextValue = itemView.textValue
        }
    }
}